File: mobile-android/app/src/main/java/com/erenlermarket/app/domain/repository/InboxRepositories.kt
package com.erenlermarket.app.domain.repository

import com.erenlermarket.app.domain.model.AppNotification
import com.erenlermarket.app.domain.model.Message
import com.erenlermarket.app.domain.model.SettingsUpdate
import com.erenlermarket.app.domain.model.UserSettings

interface MessagingRepository {
    suspend fun messages(): List<Message>
    suspend fun send(content: String): Message
}

interface NotificationsRepository {
    suspend fun notifications(): List<AppNotification>
    suspend fun markRead(id: String)
    suspend fun markAllRead()
}

interface SettingsRepository {
    suspend fun settings(): UserSettings
    suspend fun update(update: SettingsUpdate): UserSettings
}

File: mobile-ios/Modules/Domain/Sources/Repositories/InboxRepositories.swift
/// Müşteri ↔ mağaza sohbeti (tek konuşma).
public protocol MessagingRepository: Sendable {
    func messages() async throws -> [Message]
    func send(_ content: String) async throws -> Message
}

public protocol NotificationsRepository: Sendable {
    func notifications() async throws -> [AppNotification]
    func markRead(id: String) async throws
    func markAllRead() async throws
}

public protocol SettingsRepository: Sendable {
    func settings() async throws -> UserSettings
    func update(_ update: SettingsUpdate) async throws -> UserSettings
}
